src/snarks/groth16/mpc/phase2.cpp

```diff
@@ -0,0 +1,66 @@
+#include "snarks/groth16/mpc/phase2.hpp"
+
+namespace libzeth
+{
+
+template<>
+void srs_mpc_phase2_accumulator<libff::alt_bn128_pp>::write_compressed(
+    std::ostream &out) const
+{
+    using G1 = libff::alt_bn128_G1;
+    check_well_formed(*this, "mpc_layer2 (write)");
+
+    // Write fixed-sizes cs_hash and sizes first.
+
+    const size_t H_size = H_g1.size();
+    const size_t L_size = L_g1.size();
+    out.write((const char *)cs_hash, sizeof(srs_mpc_hash_t));
+    out.write((const char *)&H_size, sizeof(H_size));
+    out.write((const char *)&L_size, sizeof(L_size));
+
+    libff::alt_bn128_G1_write_compressed(out, delta_g1);
+    libff::alt_bn128_G2_write_compressed(out, delta_g2);
+    for (const G1 h : H_g1) {
+        libff::alt_bn128_G1_write_compressed(out, h);
+    }
+    for (const G1 l : L_g1) {
+        libff::alt_bn128_G1_write_compressed(out, l);
+    }
+}
+
+template<>
```

Comment:
What's the role of this?

Reply:
This provides the implementation for this specific value of `ppT` (we can only choose to write compressed / uncompressed points for `ppT == alt_bn128_pp`, so we can't provide an implementation that is generic in `ppT` without changing libff to support this for all curves)

Reply:
(To clarify, by default compression in libff is statically set with a `#define` controlled by the build config.  The functions such as `libff::alt_bn128_G1_write_compressed` were added by me in order to allow us to use both the compressed and uncompressed formats in a single build).

Reply:
I though we could just remove `template<>` because the type was specified, but the compiler is not happy with that, so let's keep it then.